nntrainer/graph/network_graph.cpp

```diff
@@ -517,14 +530,22 @@ sharedConstTensors NetworkGraph::forwarding(sharedConstTensors input) {
 
 void NetworkGraph::backwarding(sharedConstTensors output, int iteration) {
 
-  for (unsigned int i = Sorted.size() - 1; i > 0; i--) {
+  for (unsigned int i = Sorted.size() - 1; i > skip_non_trainable_layers + 1;
```

Comment:
Sorted[0] will be never accessed with this logic, I wonder if this is intended?

Reply:
Yes, sortted[0] has no one to give its derivative values to.
Further Sorted[0] is ALWAYS input layer which does not support calcGradient etc.
Hence it is always skipped.

Reply:
Got it

Reply:
done.